http-client/src/test/java/io/airlift/http/client/jetty/TestClientTracing.java

```diff
@@ -0,0 +1,124 @@
+package io.airlift.http.client.jetty;
+
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import io.airlift.bootstrap.Bootstrap;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.StringResponseHandler.StringResponse;
+import io.airlift.tracetoken.TraceTokenModule;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.handler.DefaultHandler;
+import org.eclipse.jetty.server.handler.HandlerCollection;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Qualifier;
+
+import java.io.File;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static io.airlift.http.client.HttpClientBinder.httpClientBinder;
+import static io.airlift.http.client.StringResponseHandler.createStringResponseHandler;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+import static org.eclipse.jetty.http.HttpVersion.HTTP_1_1;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+@Test(singleThreaded = true)
+public class TestClientTracing
+{
+    private Server server;
+    private Path tmpLogPath;
+
+    @BeforeMethod
+    public void setup()
+            throws Exception
+    {
+        tmpLogPath = Files.createTempDirectory("test");
+        server = new Server();
+        HttpConfiguration httpConfiguration = new HttpConfiguration();
+        ServerConnector connector;
+        connector = new ServerConnector(server, new HttpConnectionFactory(httpConfiguration));
+        connector.setIdleTimeout(60_000);
+        connector.setName("http");
+        server.addConnector(connector);
+
+        ServletContextHandler context = new ServletContextHandler();
+        context.setContextPath("/");
+        context.addServlet(TestServlet.class, "/test");
+
+        HandlerCollection handlers = new HandlerCollection();
+        handlers.setHandlers(new Handler[] {context, new DefaultHandler()});
+        server.setHandler(handlers);
+        server.start();
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void tearDown()
+            throws Exception
+    {
+        if (server != null) {
+            server.stop();
+        }
+        Files.walk(tmpLogPath)
```

Comment:
I think we have a util for this somewhere (@electrum?)

Reply:
Guava's `MoreFiles`
```java
deleteRecursively(tmpLogPath.toPath(), ALLOW_INSECURE);